Mage.Sets/src/mage/cards/x/XyrisTheWrithingStorm.java

```diff
@@ -0,0 +1,157 @@
+package mage.cards.x;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbility;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.DealsCombatDamageToAPlayerTriggeredAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.keyword.FlyingAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.token.SnakeToken;
+import mage.game.permanent.token.Token;
+import mage.players.Player;
+import mage.watchers.common.CardsDrawnDuringDrawStepWatcher;
+
+import java.util.UUID;
+
+/**
+ * @author AsterAether
+ */
+public final class XyrisTheWrithingStorm extends CardImpl {
+
+    public XyrisTheWrithingStorm(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{G}{U}{R}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.SNAKE);
+        this.subtype.add(SubType.LEVIATHAN);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(5);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // Whenever an opponent draws a card except the first one they draw in each of their draw steps, create a 1/1 green Snake creature token.
+        this.addAbility(new XyrisTheWrithingStormDrawAbility());
+        // Whenever Xyris, the Writhing Storm deals combat damage to a player, you and that player each draw that many cards.
+        this.addAbility(new DealsCombatDamageToAPlayerTriggeredAbility(new XyrisTheWrithingStormCombatDamageEffect(), false, true));
+    }
+
+    private XyrisTheWrithingStorm(final XyrisTheWrithingStorm card) {
+        super(card);
+    }
+
+    @Override
+    public XyrisTheWrithingStorm copy() {
+        return new XyrisTheWrithingStorm(this);
+    }
+}
+
+class XyrisTheWrithingStormDrawAbility extends TriggeredAbilityImpl  {
+
+    public XyrisTheWrithingStormDrawAbility() {
+        super(Zone.BATTLEFIELD, new XyrisTheWrithingStormTokenEffect(), false);
+    }
+
+    public XyrisTheWrithingStormDrawAbility(final XyrisTheWrithingStormDrawAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.DREW_CARD;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        if (game.getPlayer(this.getControllerId()).hasOpponent(event.getPlayerId(), game)) {
+            if (game.isActivePlayer(event.getPlayerId())
+                    && game.getPhase().getStep().getType() == PhaseStep.DRAW) {
+                CardsDrawnDuringDrawStepWatcher watcher = game.getState().getWatcher(CardsDrawnDuringDrawStepWatcher.class);
+                if (watcher != null && watcher.getAmountCardsDrawn(event.getPlayerId()) > 0) {
```

Comment:
Hmm, can you debug/test that code? Watchers calls first for same event (before trigger), so `getAmountCardsDrawn` must be 1 on first draw event. I see another cards with same condition (it can be bugged too -- search by `getAmountCardsDrawn`).


![shot_200418_053916](https://user-images.githubusercontent.com/8344157/79625016-037e1380-8137-11ea-9e3b-d424b907a27e.png)

Reply:
You where right, the card amount should have been 1. Fixed now.